airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -237,18 +282,23 @@ public StandardWorkspace getStandardWorkspaceFromConnection(final UUID connectio
     }
   }
 
-  public List<StandardSourceDefinition> listStandardSourceDefinitions(final boolean includeTombstone) throws JsonValidationException, IOException {
-    final List<StandardSourceDefinition> sourceDefinitions = new ArrayList<>();
-    for (final StandardSourceDefinition sourceDefinition : persistence.listConfigs(ConfigSchema.STANDARD_SOURCE_DEFINITION,
-        StandardSourceDefinition.class)) {
-      sourceDefinition.withProtocolVersion(AirbyteProtocolVersion
-          .getWithDefault(sourceDefinition.getSpec() != null ? sourceDefinition.getSpec().getProtocolVersion() : null).serialize());
-      if (!MoreBooleans.isTruthy(sourceDefinition.getTombstone()) || includeTombstone) {
-        sourceDefinitions.add(sourceDefinition);
-      }
-    }
+  public List<StandardSourceDefinition> listStandardSourceDefinitions(final boolean includeTombstone) throws IOException {
+    return sourceDefQuery(Optional.empty(), includeTombstone).toList();
+  }
 
-    return sourceDefinitions;
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private Stream<StandardSourceDefinition> sourceDefQuery(final Optional<UUID> sourceDefId, final boolean includeTombstone) throws IOException {
+    return database.query(ctx -> ctx.select(ACTOR_DEFINITION.asterisk())
+        .from(ACTOR_DEFINITION)
+        .where(ACTOR_DEFINITION.ACTOR_TYPE.eq(ActorType.source))
+        .and(sourceDefId.map(ACTOR_DEFINITION.ID::eq).orElse(noCondition()))
+        .and(includeTombstone ? noCondition() : ACTOR_DEFINITION.TOMBSTONE.notEqual(true))
+        .fetchStream())
+        .map(DbConverter::buildStandardSourceDefinition)
+        // todo(cgardens) - the impls for default protocol version were different in list and get, which is
```

Comment:
@gosusnp we have 2 different implementations of this default protocol version behavior. which is right?
1. https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java#L204-L208
2. https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java#L244-L245

Reply:
Good catch, the get is the correct behavior, we should always use the default if missing.